server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java

```diff
@@ -189,7 +187,6 @@ public HostResponse newHostResponse(HostJoinVO host, EnumSet<HostDetails> detail
         DecimalFormat decimalFormat = new DecimalFormat("#.##");
         if (host.getType() == Host.Type.Routing) {
             float cpuOverprovisioningFactor = ApiDBUtils.getCpuOverprovisioningFactor(host.getClusterId());
-            hostResponse.setCpuNumber((int)(host.getCpus() * cpuOverprovisioningFactor));
```

Comment:
this line is the main fix

the rest of the PR changes are on HostForMigrationResponse, which now extends HostResponse.
so removed lots of duplicated codes

cc @DaanHoogland @shwstppr

Reply:
OK, cpu number is already set and don't override it :+1:  (see line 123)

Reply:
yeah,exactly